website/docs/reference/resource-configs/volatility.md

```diff
@@ -0,0 +1,50 @@
+---
+title: volatility
+sidebar_label: "volatility"
+id: volatility
+---
+<VersionCallout version="1.11" /> 
+
+<File name='functions/<filename>.yml'>
+
+```yml
+functions:
+  - name: <function name>
+    config:
+      volatility: deterministic | stable | non-deterministic
+```
+
+</File>
+
+## Definition
+
+The `volatility` config is an optional config that describes how predictable a UDF’s output is. Warehouses use this to decide if results can be cached, reordered, or inlined. Setting the appropriate volatility helps prevent incorrect results when a function isn’t safe to cache or reorder.
+
+By default, dbt does not specify a volatility value. If you don’t set volatility, dbt generates a `CREATE` statement without a volatility keyword, and the warehouse’s default behavior applies &mdash; except in Redshift. In Redshift, dbt sets `non-deterministic` (`VOLATILE`) by default if no volatility is specified, because Redshift requires an explicit volatility and `VOLATILE` is the safest assumption.
+
+import Volatility from '/snippets/_warehouse-volatility.md';
+
+<Volatility />
+
+## Supported volatility types
+
+### deterministic
+
+A deterministic function always returns the same output for the same input. Because its results are predictable, the warehouse can safely apply aggressive optimizations and caching.
+
+### stable
+
+A stable function returns the same value throughout a single query execution, but its result may change across different executions. Not supported by all warehouses. For more information, see [Warehouse-specific volatility keywords](/reference/resource-configs/volatility#warehouse-specific-volatility-keywords).
```

Comment:
This could also. use an example but not sure what a good one is. Maybe current user? or query start time?